### ISSUE TITLE
net.zetetic.database.sqlcipher.SQLiteException: not authorized (code 1)

### ISSUE THREAD

**Issue by wangtao7c0** [sqlcipher/sqlcipher-android#5]
lib: "net.zetetic:sqlcipher-android:4.5.2"
`override fun postKey(connection: SQLiteConnection) {`
          ` connection.execute("SELECT load_extension('libsimple')", null, null)`
`}`

when I run this code, my app is crashed by net.zetetic.database.sqlcipher.SQLiteException: not authorized (code 1)

pls give me some advice to address this, thks

I guess, is there some compile flags not enabled?

**Comment by developernotes** [comment]
Hi @wangtao7c0 

Are you using the Commercial, or Community edition of SQLCipher for Android? The former requires that you provide a license code prior to using any encryption operation. Providing the license code can be done in the `preKey` event of the `SQLiteDatabaseHook` via:

```
SQLiteDatabaseHook hook = new SQLiteDatabaseHook() {
  public void preKey(SQLiteConnection connection) {
    connection.executeForString(String.format("PRAGMA cipher_license = '%s';", LICENSE_CODE), new Object[]{}, null);
  }
  public void postKey(SQLiteConnection connection) {}
};
```

The Community edition does not utilize a runtime licensing mechanism.

**Comment by wangtao7c0** [comment]
hi, I'm using the community edition.
`connection.execute("SELECT load_extension('libsimple')", null, null)`
is this operation must make in commercial edition?
is it possible to making it in community edition?

**Comment by developernotes** [comment]
Hi @wangtao7c0 

Both the Community and Commercial edition are built with `-DSQLITE_ENABLE_LOAD_EXTENSION` to allow for extension loading. Does SQLCipher behave properly if you remove your call to `load_extension` from the `postKey` event?

**Comment by wangtao7c0** [comment]
Yes, it work properly while removing the call to `load_extension `.
